### DISCUSSION TITLE
charge distribution/ precursor definition

### DISCUSSION THREAD

**Discussion by SSPuliasis** [vdemichev/DiaNN#667] [Q&A]
Hi,

I'm carrying out some library free searches on alternative proteases, in addition to trypsin. It is mentioned in the literature that some proteases generate peptides of a higher length and charge state. To account for this, I used a 7-35aa length range and 1-6 charge range in the in silico .speclib library generation with DIA-NN.

In the results, there are no identified precursors with a charge higher than 4. A closer look at the. speclib for Asp-N showed that all precursors in the library had charges in the 1-4 range, despite having specified a 1-6 range in .speclib generation. 
Q1: Is there a (biological or computational) reason for this? i.e. is a charge higher than +4 unheard of or impossible?

Furthermore, 2,987,351 precursors were generated for the speclib, according to DIA-NN, but the file contains 35,786,366 rows.
Q2: How exactly is a precursor defined? Many of the columns contain redundancies, but I am not sure how to filter for the 2,987,351 precursors.  

Finally, if anyone has knowledge on charge distributions for proteases other than trypsin (specifically Asp-N, Glu-C, and Chymotrypsin), it would be really useful if you could point me in the right direction

Thanks,
Sophia

**Comment by vdemichev** [comment]
Hi Sophia,

Q1: as far as I remember, the reason is that the predictor was not trained on charge > 4. 
Q2: precursor = charged peptide, each row in the library corresponds to a frament ion. No need to filter.

I would think it's unlikely that enabling higher charges will give more than 1% extra IDs with any protease.

Best,
Vadim

**Reply by vdemichev** [reply]
A quick test can be done like this: using R or Python, augment the library with charge 5 and 6 precursors, by just copying the spectra of charge 4 ones and adjusting only the respective precursor m/z values.
